### PR TITLE
Updating the documentation to include the base abstract class case in TPH

### PR DESCRIPTION
**Pull request by tinchou**
Hi!

I'm not convinced with my writing, but please feel free to use this as a starting point for a better PR.

TL;DR: there was no mention of using an abstract class in TPH.

I'm mapping a model to an existing database, and I found a very useful scenario for discriminators. We have a `Products` table, a `Licenses` table and a `ProductLicenses` table. The `Licenses` table has only six entries, and each one represents a license type with a very specific business logic.

In the old code I'm rewriting, we checked the `Id` to find out the license type and then handle it. E.g.:

``` cs
product.ProductLicenses.Any(l => l.LicenseId == Guid.Parse("....")); // if we offer this specific license type do something
```

This seemed ugly, and with EFCore I found I can strong-type this and have a simpler code. I'm setting up a discriminator like this:

``` cs
modelBuilder.Entity<ProductLicense>(entity =>
{
    entity.HasDiscriminator(e => e.LicenseId)
        .HasValue<SomeProductLicense>(Guid.Parse("F26E3C37-4709-496E-81B0-3AFFD6776D01"));
});
```

Declaring `ProductLicense` abstract (this is important! otherwise it complains you haven't set up a discriminator for the base type).

``` cs
public abstract class ProductLicense
```

And then the check is not only strongly-typed but much simpler:

``` cs
product.SomeProductLicense != null // if we offer this specific license type
```

Let me know if I can help or clarify!

### CODE REVIEW

**Comment by rowanmiller** [comment]
Using the primary key as a discriminator so that you can have a special type for one row in the database doesn't seem like a common use case, so I don't think it belongs in the documentation.
